src/content/en/updates/2018/07/page-lifecycle-api.md

```diff
@@ -0,0 +1,962 @@
+          BroadcastChannel</a> connections.</li>
+        <li>Close active <a href="https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API">
+          WebRTC</a> connections.</li>
+        <li>Stop any network polling or close any open
+          <a href="https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API">
+          Web Socket</a> connections.</li>
+        <li>Release any held <a href="https://github.com/inexorabletash/web-locks">
+          Web Locks</a>.</li>
+      </ul>
+      <p>You should also persist any dynamic view state (e.g. scroll position
+      in an infinite list view) to sessionStorage (or
+      <a href="#save-data-to-indexeddb-before-freezing">IndexedDB via
+      <code>commit()</code></a>) that you'd want restored if the page was
+      discarded and then reloaded later.</p>
+      <p>If the page then transitions from <em>frozen</em> back to
+      <em>hidden</em>, you can restore any connections or restart any polling
+      your stopped when the page was initially frozen.</p>
+   </td>
+  </tr>
+  <tr>
+    <td><strong>Terminated</strong></td>
+    <td>
+      <p>You generally do not need to take any action when a page transitions
+      to the <em>terminated</em> state.</p>
+      <p>Since pages being unloaded as a result of user action always go
+      through the <em>hidden</em> state before entering the <em>terminated</em>
+      state, the <em>hidden</em> state is where session-ending logic,
+      application state persisting, and reporting to analytics should be
+      performed.</p>
+      <p>Also (as mentioned in the <a href="#advice-hidden">recommendations for
+      the hidden state</a>), it's very important for developers to realize that
+      the transition to the terminated state cannot be reliably detected in many
+      cases (especially on mobile), so developers who depend on termination
+      events (e.g. <code>beforeunload</code>, <code>pagehide</code>, and
+      <code>unload</code>) are likely losing data.</p>
+   </td>
+  </tr>
+  <tr>
+    <td><strong>Discarded</strong></td>
+    <td>
+      <p>The <em>discarded</em> state is not observable by developers at the
+      time a page is being discarded. This is because pages are typically
+      discarded under resource constraints, and unfreezing a page just to allow
+      script to run in response to a discard event is simply not possible in
+      most cases.</p>
+      <p>As a result, you should prepare for the possibility of a discard in
+      the change from <em>hidden</em> to <em>frozen</em>, and then you can
+      react to the restoration of a discarded page at page load time by
+      checking <code>document.wasDiscarded</code>.
+    </td>
+  </tr>
+</table>
+
+## Legacy lifecycle APIs to avoid {: #legacy-lifecycle-apis-to-avoid}
+
+### The unload event {: #the-unload-event}
+
+<aside>
+  <strong>Advice summary:</strong>
+  never use the <code>unload</code> event on modern browsers
```

Comment:
'Never' should be upper case. Sentences needs a period.

Reply:
Done.